### PR TITLE
doctl: new port

### PR DESCRIPTION
**Pull request by KritR**
#### Description

doctl is a command line interface for the DigitalOcean API.

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: www/doctl/Portfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+checksums			rmd160	7d454c8812222227a28f8e4b1c27c9b37d9ef589 \
+					sha256	5f681494c3a9750e0cf944a140da2e23c117c8b8b06e474c96c73552a7162e5c \
+					size	15651051
+
+categories			www devel
+platforms			darwin
+license				Apache-2
+maintainers			{@kritr gmail.com:krdevmail} openmaintainer
+description			A command line interface for the DigitalOcean API
+long_description	${description}
```
Comment (cjones051073) [inline www/doctl/Portfile]: Minor comment, but the indentation above is a little wonky. Could you align things.

File: www/doctl/Portfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+PortSystem			1.0
+PortGroup			golang 1.0
+
+go.setup			github.com/digitalocean/doctl 1.18.0 v
+revision			0
+
+checksums			rmd160	7d454c8812222227a28f8e4b1c27c9b37d9ef589 \
+					sha256	5f681494c3a9750e0cf944a140da2e23c117c8b8b06e474c96c73552a7162e5c \
+					size	15651051
```
Comment (cjones051073) [inline www/doctl/Portfile]: Please align the indentation.

File: www/doctl/Portfile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+variant zsh_completion description {Install zsh completion} {
+	depends_run-append path:${prefix}/bin/zsh:zsh
+	post-destroot {
+		set site-functions ${destroot}${prefix}/share/zsh/site-functions
+		xinstall -d ${site-functions}
+		set completion-file [open ${site-functions}/_${name} "w"]
+		puts ${completion-file} [exec ${destroot}${prefix}/bin/${name} completion zsh]
+		close ${completion-file}
+	}
+}
```
Comment (cjones051073) [inline www/doctl/Portfile]: Any reason to not enable the bash or zsh (or both) variants by default ? 
Related question but do the zsh and bash variants conflict, i.e. can both bot be enabled at once ? If not, then they should be explicitly labelled as conflicting.

Comment (KritR) [reply www/doctl/Portfile]: Well just going through the documentation and some other ports, and I figured that leaving it to be a configurable variant is the way to go.

https://www.macports.org/ports.php?by=variant&substr=bash_completion
https://www.macports.org/ports.php?by=variant&substr=zsh_completion

Yeah it seems like there's around 14 and 8 ports that do that for each variant. And it also seems like this is a part of the official guidelines for including bash / zsh completion.

https://trac.macports.org/wiki/howto/bash-completion#Installbash_completionvariantautomaticallywithallports

However no particular reason not to enable it by default, besides not wanting to install unnecessary items onto a users filesystem.

And no the +zsh_completion and +bash_completion variants are not conflicting.

Comment (KritR) [reply www/doctl/Portfile]: Let me know if you think it should be changed from its current form.

**Pull request merged**